pkg/tablecodec/tablecodec.go

```diff
@@ -1730,8 +1719,23 @@ type IndexValueSegments struct {
 	IntHandle      []byte
 }
 
-// SplitIndexValue splits index value into segments.
+// SplitIndexValue decodes segments in index value for both non-clustered and clustered table.
 func SplitIndexValue(value []byte) (segs IndexValueSegments) {
+	if getIndexVersion(value) == 0 {
+		// For Old Encoding (IntHandle without any others options)
+		if len(value) <= MaxOldEncodeValueLen {
+			segs.IntHandle = value
+			return segs
+		}
+		// For IndexValueVersion0
+		return splitIndexValueForIndexValueVersion0(value)
+	}
+	// For IndexValueForClusteredIndexVersion1
+	return splitIndexValueForClusteredIndexVersion1(value)
+}
+
+// splitIndexValueForIndexValueVersion0 splits index value into segments.
+func splitIndexValueForIndexValueVersion0(value []byte) (segs IndexValueSegments) {
```

Comment:
not get the name `splitIndexValueForIndexValueVersion0` logic behind? this is a extensible index value format.

Reply:
See https://github.com/pingcap/tidb/blob/2d552c00343ca9a457fcda2924c3940f7f46add0/pkg/tablecodec/tablecodec.go#L1408, we have three different index value types.

Reply:
okok